diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..be9e7e5
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,52 @@
+
+project ( lua C )
+cmake_minimum_required ( VERSION 2.8 )
+
+# Sources
+set (SRC_CORE
+  lzio.c
+  lctype.c
+  lopcodes.c
+  lmem.c
+  lundump.c
+  ldump.c
+  lstate.c
+  lgc.c
+  llex.c
+  lcode.c
+  lparser.c
+  ldebug.c
+  lfunc.c
+  lobject.c
+  ltm.c
+  lstring.c
+  ltable.c
+  ldo.c
+  lvm.c
+  lapi.c
+)
+source_group ("Source Files\\Core" FILES ${SRC_ImGui})
+
+set (SRC_LIB
+  lauxlib.c
+  lbaselib.c
+  lcorolib.c
+  ldblib.c
+  liolib.c
+  lmathlib.c
+  loadlib.c
+  loslib.c
+  lstrlib.c
+  ltablib.c
+  lutf8lib.c
+  linit.c
+)
+source_group ("Source Files\\Library" FILES ${SRC_ImGui})
+
+## BUILD
+add_library (liblua ${SRC_CORE} ${SRC_LIB})
+target_include_directories (liblua PUBLIC .)
+set_target_properties(liblua PROPERTIES OUTPUT_NAME "lua")
+
+add_executable (lua lua.c)
+target_link_libraries(lua PUBLIC liblua)
\ No newline at end of file
diff --git a/lua.hpp b/lua.hpp
new file mode 100644
index 0000000..f1b6bfa
--- /dev/null
+++ b/lua.hpp
@@ -0,0 +1,9 @@
+// lua.hpp
+// Lua header files for C++
+// <<extern "C">> not supplied automatically because Lua also compiles as C++
+
+extern "C" {
+#include "lua.h"
+#include "lualib.h"
+#include "lauxlib.h"
+}
\ No newline at end of file
